posts/2023-02-15-run-specific-tests-with-exunit.md

```diff
@@ -0,0 +1,93 @@
+%{
+  author: "Kevin Mathew",
+  author_link: "https://github.com/kevinam99",
+  tags: ["testing", "ExUnit"],
+  date: ~D[2023-02-15],
+  title: "Run specific test cases with ExUnit",
+  excerpt: """
+  Learn how you can easily run one specific test case or a group test cases in Elixir without having to run through the whole test suites and custom workarounds.
+  """
+}
+---
+
+When starting with Elixir, I was amazed how easy it was to write tests in Elixir with ExUnit and be able to do true TDD. This was fine till I was working on my own projects where the size was small and didn't have many components. But as I got to working full time, it became difficult to test specific functionalities that I wrote in a big codebase. 
+
+I found myself commenting out tests that I want to skip which made the whole process tedious and I was back in iex testing functions manually. 
+My woes ended when I found these two ways of running specific test cases and I hope this helps as much as it helped me.
+
+Imagine a test file with several tests but you want to run the test case only for the `add/2` function that simply adds two numbers and gives a result.
+
+```elixir
+test "it adds two numbers correctly" do
+  assert add(2, 2) == 4
+end
+
+```
+
+There are three ways to go about this:
+### 1. Specify the line number of the test in the `mix test` command.
+If the above test case is on, say, line 100 in the test file, then run your command as
+
+```shell
+$ mix test ./path/to/test/file/math_test.exs:100
+
+```
+  This will run the test case on line 100 and skip all other tests. Of course, the basic setup will be done. Boom! Saved so much time and back to being productive.
```

Comment:
```suggestion
This will run the test case on line 100 and skip all other tests. Of course, the basic setup will be done. Boom! Saved so much time and back to being productive.
```